ostd/src/arch/x86/mod.rs

```diff
@@ -79,6 +82,25 @@ pub fn read_tsc() -> u64 {
     unsafe { _rdtsc() }
 }
 
+/// Reads a hardware generated 64-bit random value.
+///
+/// Returns None if no random value was generated.
+pub fn rdrand() -> Option<u64> {
```

Comment:
To comply with Rust naming style (rather Intel's instruction naming style), I suggest naming this function `read_random`. This function is name consistent with the name of `read_tsc` in the same module.